component/component.go

```diff
@@ -79,6 +80,10 @@ type Host interface {
 	// thus we may have an instance of the exporter for multiple data types.
 	// This is an experimental function that may change or even be removed completely.
 	GetExporters() map[configmodels.DataType]map[configmodels.Exporter]Exporter
+
+	// Return the resource information that was automatically detected as being associated
+	// with this host based on the configured resource detectors.
+	GetResource() pdata.Resource
```

Comment:
I'd like to block this until discussion here is resolved https://github.com/open-telemetry/opentelemetry-collector/issues/871

Reply:
👍  This was just a PoC to encourage discussion. I'll close this PR for now and create a new one later based on a different approach.